### PR TITLE
feat: Support OpenVLA/OFT modified LIBERO RLDS datasets for training

### PR DESCRIPTION
**Pull request by VagRant2333**
## Motivation and Context

This PR adds support for the OpenVLA / OpenVLA-OFT modified LIBERO RLDS datasets (`openvla/modified_libero_rlds`), including the commonly used task-suite subsets `libero_spatial_no_noops`, `libero_object_no_noops`, `libero_goal_no_noops`, and `libero_10_no_noops`.

This dataset family is the LIBERO fine-tuning path used in [OpenVLA](https://github.com/openvla/openvla) and carried forward by [OpenVLA-OFT](https://github.com/moojink/openvla-oft). It is also adopted or explicitly referenced by several follow-up open-source VLA codebases, including [VLA-Adapter](https://github.com/OpenHelix-Team/VLA-Adapter), [LightVLA](https://github.com/LiAutoAD/LightVLA/blob/main/LIBERO.md), [CogVLA](https://github.com/JiuTian-VL/CogVLA/blob/main/docs/LIBERO.md), [UniVLA](https://github.com/OpenDriveLab/UniVLA), and [RD-VLA](https://github.com/rd-vla/rd-vla).

Supporting this dataset family in starVLA reduces the gap between the current LeRobot-based LIBERO pipeline and a widely used OpenVLA-style training setup, improving reproducibility and lowering friction for comparing against prior and follow-up OFT-style baselines.

## Key Changes

- **RLDS dataloader adapter (`rlds_libero_datasets.py`)**: Added an `IterableDataset` to stream RLDS data into PyTorch efficiently. It handles action chunking (horizon padding), state precision formatting, and multi-node sharding in a way that is consistent with `starVLA/dataloader/lerobot_datasets.py`.
- **Dependency decoupling**: Updated `starVLA/dataloader/__init__.py` so that the RLDS-specific adapter is imported lazily. This keeps the existing LeRobot-only workflow unchanged and avoids introducing RLDS dependencies into code paths that do not use this dataset backend.
- **Data preparation and inspection**:
  - Added `data_preparation_rlds.sh` as a lightweight helper for preparing the expected local dataset layout from the Hugging Face dataset source.
  - Added validation utilities (`validate_rlds_statistics.py` and `inspect_rlds_orientation.py`) to inspect dataset statistics and verify image orientation for OpenVLA-style modified LIBERO RLDS datasets, since this dataset family requires dataset-specific handling during conversion and loading.
- **Config and entrypoints**: Added an example config (`starvla_cotrain_libero_rlds.yaml`) and a convenience launch script (`run_libero_train_rlds.sh`) for the new data path.

## Experimental Validation

I ran internal integration checks with this data path and confirmed that the training and evaluation pipeline complete normally on my side with current starVLA codebase.

## Usage

```bash
# 1. Prepare data
export DEST=/path/to/data
bash examples/LIBERO/data_preparation_rlds.sh

# 2. Start training
bash examples/LIBERO/train_files/run_libero_train_rlds.sh
```

### CODE REVIEW

**Comment by JinhuiYE** [comment]
Hi,

Thanks a lot for this detailed PR — really appreciate the effort you put into adding OpenVLA-style RLDS dataset support.

We do have one concern: supporting two different data formats (LeRobot and RLDS) in parallel might increase the engineering complexity of the codebase, making it harder to maintain and debug.

For that reason, we'd prefer to recommend using the conversion tool at https://github.com/Tavish9/any4lerobot/tree/main/libero2lerobot to convert LIBERO datasets into the standard LeRobot format, which is already supported by starVLA.

That said, your PR is still a valuable contribution. We'll make sure to highlight your work in our updates and acknowledge your effort properly.

Thanks again for helping move the community forward!

**Comment by VagRant2333** [comment]
Thanks a lot for the thoughtful feedback and i understand that  maintaining two dataset formats is inapropriate.

In that case, i am happy to align with that direction. I’ll treat this RLDS support work as a reference implementation on my side, and I can instead help improve the conversion-based workflow around modified libero datasets.

Concretely, I can look into contributing to clearer documentation for converting `openvla/modified_libero_rlds` into LeRobot format for starVLA

If helpful, I’d also be glad to open a lighter PR for documentation or user guidance around this conversion workflow.

Thanks again for your kind words.

**Comment by JinhuiYE** [comment]
Thanks again for your understanding and willingness to align on this direction.

I really appreciate you taking the time to treat the RLDS support as a reference implementation and to focus on improving the conversion-based workflow for modified LIBERO datasets.

Your plan to contribute clearer documentation — especially on converting openvla/modified_libero_rlds into LeRobot format for StarVLA — sounds great.

Please let me know once you’ve made progress, and feel free to open a lighter PR for documentation or user guidance around the conversion workflow. That would be super helpful for others who want to follow your steps.

Looking forward to your contribution!
